src/include/daos_srv/pool.h

```diff
@@ -292,10 +292,11 @@ ds_pool_rf_verify(struct ds_pool *pool, uint32_t last_ver, uint32_t rlvl, uint32
 static inline uint32_t
 ds_pool_get_version(struct ds_pool *pool)
 {
-	uint32_t	ver;
+	uint32_t	ver = 0;
 
 	ABT_rwlock_rdlock(pool->sp_lock);
-	ver = pool_map_get_version(pool->sp_map);
+	if (pool->sp_map != NULL)
+		ver = pool_map_get_version(pool->sp_map);
```

Comment:
[Note] I'm considering to redesign the management of ds_pool objects with a new "status" field, so that we'll be able to hide a ds_pool object who hasn't been fully ready yet. Then most of the code will be able to assume that sp_map is always valid, like sp_group. Hence, the current patch serves as a short-term fix, for this particular segfault seems easy to reproduce.

Reply:
return 0 is a bit obscure, probably let the caller check sp_map explicitly is clearer? Though given the initial pool map version is 1, probably return 0 is ok.

Reply:
@wangdi1, my original design of the "remove orterun dependency" was to let every user of sp_map to check if it is NULL. Because of bugs like this, now I think that was probably bad decision. Hence the [Note] comment above.

We already use 0 to indicate an uninitialized map elsewhere in the code, I think. Since 0 < any valid map version, it seems to work consistently.